ci/docker/conda-python.dockerfile

```diff
@@ -22,13 +22,15 @@ FROM ${org}/${arch}-conda-cpp:latest
 # install python specific packages
 ARG python=3.6
 COPY ci/conda_env_python.yml /arrow/ci/
-RUN conda install -q \
+RUN conda install -n testenv -q \
         --file arrow/ci/conda_env_python.yml \
         python=${python} \
-        tensorflow \
         nomkl && \
     conda clean --all
 
+COPY python/requirements.txt /arrow/python/
+RUN bash -c "source activate testenv && pip install -r /arrow/python/requirements.txt"
```

Comment:
Why is this needed? 
If there is something missing (six?), should that be added to `conda_env_python.yml` ?

Reply:
It's needed for 2.7 backports such as [futures](https://pypi.org/project/futures/). They must only be installed on Python 2 builds, and I don't know how to express that in a conda YAML file.

Reply:
We should find a better way to install the dependencies for python 2.